plugins/tekton/src/utils/tekton-utils.ts

```diff
@@ -0,0 +1,203 @@
+  const skippedTaskLength = pipelinerun?.status?.skippedTasks?.length || 0;
+  const taskStatus: TaskStatus = {
+    PipelineNotStarted: 0,
+    Pending: 0,
+    Running: 0,
+    Succeeded: 0,
+    Failed: 0,
+    Cancelled: 0,
+    Skipped: skippedTaskLength,
+  };
+  if (!pipelinerun?.status?.taskRuns) {
+    return taskStatus;
+  }
+  const plrTasks = Object.keys(pipelinerun.status.taskRuns);
+  plrTasks.forEach((taskRun: string) => {
+    const taskRunStatus = pipelinerun?.status?.taskRuns?.[taskRun];
+    const status = taskRunStatus && pipelineRunFilterReducer(taskRunStatus);
+    if (status === 'Succeeded') {
+      taskStatus[ComputedStatus.Succeeded]++;
+    } else if (status === 'Running') {
+      taskStatus[ComputedStatus.Running]++;
+    } else if (status === 'Failed') {
+      taskStatus[ComputedStatus.Failed]++;
+    } else if (status === 'Cancelled') {
+      taskStatus[ComputedStatus.Cancelled]++;
+    } else {
+      taskStatus[ComputedStatus.Pending]++;
+    }
+  });
+
+  return {
+    ...taskStatus,
+  };
+};
+
+export const getTaskStatus = (pipelinerun: PipelineRunKind) => {
+  const totalTasks = totalPipelineRunTasks(pipelinerun);
+  const plrTasks = pipelinerun?.status?.taskRuns
+    ? Object.keys(pipelinerun.status.taskRuns)
+    : [];
+  const plrTaskLength = plrTasks.length;
+  const skippedTaskLength = pipelinerun?.status?.skippedTasks?.length || 0;
+
+  const taskStatus: TaskStatus = updateTaskStatus(pipelinerun);
+
+  if (pipelinerun?.status?.taskRuns) {
+    const pipelineRunHasFailure = taskStatus[ComputedStatus.Failed] > 0;
+    const pipelineRunIsCancelled =
+      pipelineRunFilterReducer(pipelinerun) === ComputedStatus.Cancelled;
+    const unhandledTasks =
+      totalTasks >= plrTaskLength
+        ? totalTasks - plrTaskLength - skippedTaskLength
+        : totalTasks;
+
+    if (pipelineRunHasFailure || pipelineRunIsCancelled) {
+      taskStatus[ComputedStatus.Cancelled] += unhandledTasks;
+    } else {
+      taskStatus[ComputedStatus.Pending] += unhandledTasks;
+    }
+  } else if (
+    pipelinerun?.status?.conditions?.[0]?.status === 'False' ||
+    pipelinerun?.spec.status === SucceedConditionReason.PipelineRunCancelled
+  ) {
+    taskStatus[ComputedStatus.Cancelled] = totalTasks;
+  } else if (
+    pipelinerun?.spec.status === SucceedConditionReason.PipelineRunPending
+  ) {
+    taskStatus[ComputedStatus.Pending] += totalTasks;
+  } else {
+    taskStatus[ComputedStatus.PipelineNotStarted]++;
+  }
+  return taskStatus;
+};
+
+export const getDuration = (seconds: number, long?: boolean): string => {
+  if (seconds === 0) {
+    return 'less than a sec';
+  }
+  let sec = Math.round(seconds);
+  let min = 0;
+  let hr = 0;
+  let duration = '';
+  if (sec >= 60) {
+    min = Math.floor(sec / 60);
+    sec %= 60;
+  }
+  if (min >= 60) {
+    hr = Math.floor(min / 60);
+    min %= 60;
+  }
+  if (hr > 0) {
+    duration += long ? `${hr} hour` : `${hr}h`;
```

Comment:
Use the `pluralize` util from @patternfly/react-core

```suggestion
    duration += long ? pluralize(hr, 'hour', 'hours') : `${hr}h`;
```

![Uploading Screenshot 2023-04-10 at 11.54.16 PM.png…]()